src/java/com/google/cloud/bigquery/dwhassessment/hooks/test_utils/TestUtils.java

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery.dwhassessment.hooks.test_utils;
+
+import static com.google.cloud.bigquery.dwhassessment.hooks.logger.LoggingHookConstants.QUERY_EVENT_SCHEMA;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericData.Record;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.avro.io.DatumReader;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.QueryState;
+import org.apache.hadoop.hive.ql.hooks.HookContext;
+import org.apache.hadoop.hive.ql.log.PerfLogger;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.parse.BaseSemanticAnalyzer;
+import org.apache.hadoop.hive.ql.parse.DDLSemanticAnalyzer;
+import org.apache.hadoop.hive.ql.plan.HiveOperation;
+
+/** Common utils for testing */
+public final class TestUtils {
+
+  public static final long QUERY_END_TIME = 9999L;
+  public static final String DEFAULT_QUERY_TEXT = "SELECT * FROM employees";
+  public static final String DEFAULT_QUERY_ID = "hive_query_id_999";
+
+  private final HiveConf conf;
+  private final QueryState queryState;
+  private final QueryPlan queryPlan;
+  private final HookContext hookContext;
+
+  public TestUtils(Hive hiveMock) throws Exception {
+    conf = new HiveConf();
+    queryState = new QueryState(conf);
+    BaseSemanticAnalyzer sem = new DDLSemanticAnalyzer(queryState, hiveMock);
+    queryPlan = new QueryPlan(DEFAULT_QUERY_TEXT, sem, 1234L, DEFAULT_QUERY_ID,
+        HiveOperation.QUERY, null);
+    PerfLogger perfLogger = PerfLogger.getPerfLogger(conf, true);
+    hookContext = new HookContext(
+        queryPlan,
+        queryState,
+        null,
+        "test_user",
+        "192.168.10.10",
+        "hive_addr",
+        "test_op_id",
+        "test_session_id",
+        "test_thread_id",
+        true,
+        perfLogger);
+  }
+
+  public QueryState getQueryState() {
+    return queryState;
+  }
+
+  public QueryPlan getQueryPlan() {
+    return queryPlan;
+  }
+
+  public HookContext getHookContext() {
+    return hookContext;
+  }
+
+  public HiveConf getConf() {
+    return conf;
+  }
+
+  public static Clock getFixedClock() {
+    return Clock.fixed(Instant.ofEpochMilli(QUERY_END_TIME), ZoneOffset.UTC);
+  }
+
+  public static Record getPreExecRecord() {
```

Comment:
I would suggest using `create` instead of `get` since the latter could be understood as a retrieval process of the saved variable. Compare to `getHookContext`.

Ditto for `getFixedClock` and `getPostExecRecord`